datalab/datalab_session/data_operations/data_operation.py

```diff
@@ -99,42 +99,54 @@ def set_failed(self, message: str):
         self.set_status('FAILED')
         self.set_message(message)
 
-    # percent lets you allocate a fraction of the operation that this takes up in time
-    # cur_percent is the current completion of the operation
-    def create_and_store_fits(self, hdu_list: fits.HDUList, percent=None, cur_percent=None) -> list:
-        if not type(hdu_list) == list:
-            hdu_list = [hdu_list]
+    def create_jpg_output(self, fits_paths, percent=None, cur_percent=None, color=False, index=None) -> list:
+        """
+        Create jpgs from fits files and save them to S3
+        If using the color option fits_paths need to be in order R, G, B
+        percent and cur_percent are used to update the progress of the operation
+        """
 
-        output = []
-        total_files = len(hdu_list)
+        if type(fits_paths) is not list:
+            fits_paths = [fits_paths]
 
-        # Create temp file paths for storing the products
-        fits_path           = tempfile.NamedTemporaryFile(suffix=f'{self.cache_key}.fits').name
+        # create the jpgs from the fits files
         large_jpg_path      = tempfile.NamedTemporaryFile(suffix=f'{self.cache_key}-large.jpg').name
         thumbnail_jpg_path  = tempfile.NamedTemporaryFile(suffix=f'{self.cache_key}-small.jpg').name
 
-        for index, hdu in enumerate(hdu_list, start=1):
-            height, width = hdu[1].shape
+        height, width = fits_dimensions(fits_paths[0])
```

Comment:
Do we assume all dimensions are the same? Or do we throw some sort of exception if they are? This would be a good unit test case.

Reply:
fits2img can handle cropping images if they're not the same width/height so I went with the first image, to set as a target size for the jpg.

Reply:
probably should go with the max width,height of all the images then